directory-connector/src/com/untangle/app/directory_connector/GroupManager.java

```diff
@@ -496,21 +496,21 @@ public void run()
             logger.info("Renewing AD Group Cache...");
             long startTime = System.currentTimeMillis();
 
-            List<String> domains = null;
+            List<ActiveDirectoryLdapAdapter> adAdapters = null;
             try {
-                domains = app.getActiveDirectoryManager().getDomains();
+                adAdapters = app.getActiveDirectoryManager().getAdapters();
             } catch ( Exception ex ) {
                 logger.warn("Unable to retrieve the domains", ex);
                 return;
             }
 
-            Map<String,Map<String,Map<String,Boolean>>> domainsGroupsUsersCache = new ConcurrentHashMap<>(domains.size());
-            Map<String,Map<String,Set<String>>> domainsGroupsChildrenCache = new ConcurrentHashMap<>(domains.size());
-            Map<String,Map<String,Boolean>> domainsUsersCache = new ConcurrentHashMap<>(domains.size());
+            Map<String,Map<String,Map<String,Boolean>>> domainsGroupsUsersCache = new ConcurrentHashMap<>(adAdapters.size());
```

Comment:
lists's `size()` function could be expensive, lets not call it repeatedly.
Use an `int` to store the size and use it everywhere.